spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/NamedConfigMapContextToSourceDataProvider.java

```diff
@@ -80,59 +71,41 @@ public KubernetesClientContextToSourceData get() {
 		return context -> {
 
 			NamedConfigMapNormalizedSource source = (NamedConfigMapNormalizedSource) context.normalizedSource();
-			// namespace has to be read from context, not from the normalized source
 			String namespace = context.namespace();
-			Environment environment = context.environment();
-			String configMapName = appName(environment, source).get();
+			String initialConfigMapName = appName(context.environment(), source).get();
+			String currentConfigMapName = initialConfigMapName;
 			Set<String> propertySourceNames = new LinkedHashSet<>();
-			propertySourceNames.add(configMapName);
+			propertySourceNames.add(initialConfigMapName);
 
 			Map<String, Object> result = new HashMap<>();
+
+			LOG.info("Loading ConfigMap with name '" + initialConfigMapName + "' in namespace '" + namespace + "'");
 			try {
-				Set<String> names = new HashSet<>();
-				names.add(configMapName);
-				if (environment != null && source.profileSpecificSources()) {
-					for (String activeProfile : environment.getActiveProfiles()) {
-						names.add(configMapName + "-" + activeProfile);
+				Map<String, String> data = getConfigMapData(context.client(), namespace, currentConfigMapName);
+				result.putAll(withPrefix(entriesProcessor.apply(data, context.environment()), source.prefix(), ""));
```

Comment:
get the data from a non-profile based config map and then, on the result, call `withPrefix`. Same is being done for profile-based sources

Reply:
this method is identical to the fabric8 one, with a different implementation of `getConfigMapData`, only.